planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_searcher.cpp

```diff
@@ -93,13 +94,59 @@ struct SortByWeightedDistance
 };
 
 GoalSearcher::GoalSearcher(
-  const GoalPlannerParameters & parameters, const LinearRing2d & vehicle_footprint)
-: GoalSearcherBase{parameters},
-  vehicle_footprint_{vehicle_footprint},
-  left_side_parking_{parameters.parking_policy == ParkingPolicy::LEFT_SIDE}
+  const GoalPlannerParameters & parameters, const LinearRing2d & vehicle_footprint,
+  const bool left_side_parking, const lanelet::ConstLanelets & pull_over_lanes,
+  const lanelet::BasicPolygons2d & no_parking_area_polygons,
+  const lanelet::BasicPolygons2d & no_stopping_area_polygons,
+  const lanelet::BasicPolygons2d & bus_stop_area_polygons)
+: parameters_(parameters),
+  vehicle_footprint_(vehicle_footprint),
+  left_side_parking_(left_side_parking),
+  pull_over_lanes_(pull_over_lanes),
+  no_parking_area_polygons_(no_parking_area_polygons),
+  no_stopping_area_polygons_(no_stopping_area_polygons),
+  bus_stop_area_polygons_(bus_stop_area_polygons)
 {
 }
 
+GoalSearcher GoalSearcher::create(
+  const GoalPlannerParameters & parameters, const LinearRing2d & vehicle_footprint,
+  const std::shared_ptr<const PlannerData> & planner_data)
```

Comment:
(minor:
just route_handler is enough. 
and in all goal searcher functions, route_handler and paramter are enough.)